src/VisualStudio/Core/Def/ErrorReporting/VisualStudioInfoBar.cs

```diff
@@ -4,119 +4,174 @@
 
 using System;
 using System.Collections.Generic;
-using System.Diagnostics.CodeAnalysis;
 using System.Linq;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
 using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
 using Microsoft.CodeAnalysis.Shared.TestHooks;
 using Microsoft.VisualStudio;
 using Microsoft.VisualStudio.Imaging;
+using Microsoft.VisualStudio.Imaging.Interop;
 using Microsoft.VisualStudio.Shell;
 using Microsoft.VisualStudio.Shell.Interop;
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.ErrorReporting;
 
-internal sealed class VisualStudioInfoBar
+internal sealed class VisualStudioInfoBar(
+    IThreadingContext threadingContext,
+    SVsServiceProvider serviceProvider,
```

Comment:
💡 This parameter should be removed, and any necessary `IVsService<TInterface>` members added instead. Note that for MEF _constructors_, you'll see `IVsService<TService, TInterface>` but when those values are stored and passed on, it'll be done as the base interface `IVsService<TInterface>` for simplicity.